app/Http/Controllers/ScoresController.php

```diff
@@ -22,6 +22,7 @@ public function __construct()
 
         $this->middleware('auth', ['except' => [
             'show',
+            'download',
```

Comment:
this will cause the files to be downloadable without logging in on web which I don't think is an intended side effect.

also the current function assumes current user exists and thus it will throw error when the user is null.

the function needs to be adjusted to handle those cases.

Reply:
That is basically what I was asking on the Discord, since I don't know how I would 1. allow client credential auth and 2. check for such auth to not run the code getting the resource owner and increasing the view count. I couldn't quite figure out how to based on what you said.

Reply:
if you look at the actual function, there's a variable which stores the current user which will be null if accessed through web without logging in or client credentials.

As for api vs web check, there's a function being called in the function which should be relatively obvious on what it does.

Reply:
If accessed through client credential auth, shouldn't the user still be null? Since client credential does not have a resource owner. Because of that I thought there's nothing to do there.

Reply:
In that case, I don't seem to understand what code even disallows client-credential auth. `is_api_request()` is true so it won't redirect. Having to handle nullability on the user object makes sense

Reply:
it's this auth middleware. but as I mentioned above, the download function needs to be adjusted to handle the change this caused